### DISCUSSION TITLE
NodeClass Managers in Milo

### DISCUSSION THREAD

**Discussion by ArnoLWeiss** [eclipse-milo/milo#786] [General]
I am interested in getting all available Types from a server to a client application. There's three implementations of managers for NodeClasses:

- [ObjectTypeManager](https://github.com/eclipse/milo/blob/f581575cfbf53e0b7d4528ef0bbd515cf7781930/opc-ua-sdk/sdk-client/src/main/java/org/eclipse/milo/opcua/sdk/client/ObjectTypeManager.java#L28)
- [DataTypeManager](https://github.com/eclipse/milo/blob/master/opc-ua-stack/stack-core/src/main/java/org/eclipse/milo/opcua/stack/core/types/DataTypeManager.java)
- [VariableTypeManager](https://github.com/eclipse/milo/blob/master/opc-ua-sdk/sdk-client/src/main/java/org/eclipse/milo/opcua/sdk/client/VariableTypeManager.java)

I had hoped that all would implement a `getXyzTypeDictionary`-Method, however, only the DataTypeManager does. Is this by design?

**Comment by kevinherron** [comment]
I don’t think these classes do what you think they do. Further, there is no type dictionary for anything but data types.
